23년도 파일만 사용하면 됨, 1970~2023년 데이터 다 있음

In [4]:
from google.colab import files

uploaded = files.upload()   # 지역별 선박입출항 실적 통계정보 22~24년도 불러옴 - 창원 항만권역의 물류 수요 증가 알아보기 위헤

Saving 산업통상자원부_자유무역지역 수출입실적 현황_20231231.csv to 산업통상자원부_자유무역지역 수출입실적 현황_20231231.csv


In [5]:
from google.colab import drive
drive.mount('/content/drive')

import os

base_path = "/content/drive/MyDrive/contest_changwon_port"

folders = [
    "data/raw/port_calls",
    "data/raw/port_facility",
    "data/raw/ship_spec",
    "data/raw/trucking_company",
    "data/raw/freight_forwarder",
    "data/processed",
    "output/tables",
    "output/figures"
]

for folder in folders:
    path = os.path.join(base_path, folder)
    os.makedirs(path, exist_ok=True)
    print("생성/확인 완료:", path)

Mounted at /content/drive
생성/확인 완료: /content/drive/MyDrive/contest_changwon_port/data/raw/port_calls
생성/확인 완료: /content/drive/MyDrive/contest_changwon_port/data/raw/port_facility
생성/확인 완료: /content/drive/MyDrive/contest_changwon_port/data/raw/ship_spec
생성/확인 완료: /content/drive/MyDrive/contest_changwon_port/data/raw/trucking_company
생성/확인 완료: /content/drive/MyDrive/contest_changwon_port/data/raw/freight_forwarder
생성/확인 완료: /content/drive/MyDrive/contest_changwon_port/data/processed
생성/확인 완료: /content/drive/MyDrive/contest_changwon_port/output/tables
생성/확인 완료: /content/drive/MyDrive/contest_changwon_port/output/figures


In [6]:
free_trade_path = os.path.join(base_path, "data/raw/free_trade_export_import")
os.makedirs(free_trade_path, exist_ok=True)

print("폴더 생성/확인 완료:", free_trade_path)

폴더 생성/확인 완료: /content/drive/MyDrive/contest_changwon_port/data/raw/free_trade_export_import


In [7]:
import pandas as pd
import os
import glob

free_trade_path = os.path.join(base_path, "data/raw/free_trade_export_import")

files = glob.glob(os.path.join(free_trade_path, "*20231231*.csv"))

print("선택된 파일 수:", len(files))
for f in files:
    print(os.path.basename(f))

선택된 파일 수: 1
산업통상자원부_자유무역지역 수출입실적 현황_20231231.csv


In [8]:
# 자유무역지역 파일 불러오기

free_trade_file = files[0]

free_trade = pd.read_csv(free_trade_file, encoding="cp949")

print("데이터 크기:", free_trade.shape)
display(free_trade.head())
print(free_trade.columns)

데이터 크기: (54, 36)


,연도,마산_수출실적(천달러),마산_수입실적(천달러),마산_무역수지(천달러),마산_고용인원,마산_업체수,대불_수출실적(천달러),대불_수입실적(천달러),대불_무역수지(천달러),대불_고용인원,...,군산_수출실적(천달러),군산_수입실적(천달러),군산_무역수지(천달러),군산_고용인원,군산_업체수,동해_수출실적(천달러),동해_수입실적(천달러),동해_무역수지(천달러),동해_고용인원,동해_업체수
0,1970,NaN,NaN,NaN,NaN,4,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,1971,857.0,618.0,239.0,1248.0,22,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,1972,9739.0,6731.0,3008.0,7106.0,70,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,1973,70374.0,51631.0,18743.0,21240.0,115,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,1974,181547.0,153600.0,27947.0,20822.0,110,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


Index(['연도', '마산_수출실적(천달러)', '마산_수입실적(천달러)', '마산_무역수지(천달러)', '마산_고용인원',
       '마산_업체수', '대불_수출실적(천달러)', '대불_수입실적(천달러)', '대불_무역수지(천달러)', '대불_고용인원',
       '대불_업체수', '율촌_수출실적(천달러)', '율촌_수입실적(천달러)', '율촌_무역수지(천달러)', '율촌_고용인원',
       '율촌_업체수', '김제_수출실적(천달러)', '김제_수입실적(천달러)', '김제_무역수지(천달러)', '김제_고용인원',
       '김제_업체수', '울산_수출실적(천달러)', '울산_수입실적(천달러)', '울산_무역수지(천달러)', '울산_고용인원',
       '울산_업체수', '군산_수출실적(천달러)', '군산_수입실적(천달러)', '군산_무역수지(천달러)', '군산_고용인원',
       '군산_업체수', '동해_수출실적(천달러)', '동해_수입실적(천달러)', '동해_무역수지(천달러)', '동해_고용인원',
       '동해_업체수'],
      dtype='object')


In [9]:
# 마산 자유무역지역만 추출

masan_free_trade = free_trade[
    [
        "연도",
        "마산_수출실적(천달러)",
        "마산_수입실적(천달러)",
        "마산_무역수지(천달러)",
        "마산_고용인원",
        "마산_업체수"
    ]
].copy()

display(masan_free_trade.tail())

,연도,마산_수출실적(천달러),마산_수입실적(천달러),마산_무역수지(천달러),마산_고용인원,마산_업체수
49,2019,784791.0,191152.0,593579.0,5400.0,115
50,2020,768680.0,203814.0,564658.0,5328.0,120
51,2021,947255.0,238914.0,708341.0,5314.0,126
52,2022,987303.0,228764.0,758539.0,5625.0,127
53,2023,1056143.0,247410.0,808734.0,5662.0,131


원래는 22-24년도 데이터 보고 싶은데 현재 파일은 23년까지 있기 때문에 20~23 추세 + 22~23 최근 변화 사용

In [10]:
masan_recent = masan_free_trade[
    masan_free_trade["연도"].between(2020, 2023)
].copy()

display(masan_recent)

,연도,마산_수출실적(천달러),마산_수입실적(천달러),마산_무역수지(천달러),마산_고용인원,마산_업체수
50,2020,768680.0,203814.0,564658.0,5328.0,120
51,2021,947255.0,238914.0,708341.0,5314.0,126
52,2022,987303.0,228764.0,758539.0,5625.0,127
53,2023,1056143.0,247410.0,808734.0,5662.0,131


In [15]:
# 총교역액 만들기

masan_recent["마산_총교역액(천달러)"] = (
    masan_recent["마산_수출실적(천달러)"] +
    masan_recent["마산_수입실적(천달러)"]
)

display(masan_recent)


#################################################

# 전년 대비 증가율 계산

masan_recent = masan_recent.sort_values("연도").reset_index(drop=True)

growth_cols = [
    "마산_수출실적(천달러)",
    "마산_수입실적(천달러)",
    "마산_총교역액(천달러)",
    "마산_고용인원",
    "마산_업체수"
]

for col in growth_cols:
    masan_recent[col + "_전년대비증가율"] = masan_recent[col].pct_change() * 100

display(masan_recent)

,연도,마산_수출실적(천달러),마산_수입실적(천달러),마산_무역수지(천달러),마산_고용인원,마산_업체수,마산_총교역액(천달러)
50,2020,768680.0,203814.0,564658.0,5328.0,120,972494.0
51,2021,947255.0,238914.0,708341.0,5314.0,126,1186169.0
52,2022,987303.0,228764.0,758539.0,5625.0,127,1216067.0
53,2023,1056143.0,247410.0,808734.0,5662.0,131,1303553.0


,연도,마산_수출실적(천달러),마산_수입실적(천달러),마산_무역수지(천달러),마산_고용인원,마산_업체수,마산_총교역액(천달러),마산_수출실적(천달러)_전년대비증가율,마산_수입실적(천달러)_전년대비증가율,마산_총교역액(천달러)_전년대비증가율,마산_고용인원_전년대비증가율,마산_업체수_전년대비증가율
0,2020,768680.0,203814.0,564658.0,5328.0,120,972494.0,NaN,NaN,NaN,NaN,NaN
1,2021,947255.0,238914.0,708341.0,5314.0,126,1186169.0,23.231384,17.221584,21.971858,-0.262763,5.000000
2,2022,987303.0,228764.0,758539.0,5625.0,127,1216067.0,4.227795,-4.248391,2.520551,5.852465,0.793651
3,2023,1056143.0,247410.0,808734.0,5662.0,131,1303553.0,6.972530,8.150758,7.194176,0.657778,3.149606


2020년 기준 지수 만들기 - 20년 대비 얼마나 커졋냐 보는 거

In [16]:
base_year = 2020

for col in growth_cols:
    base_value = masan_recent.loc[
        masan_recent["연도"] == base_year, col
    ].values[0]

    masan_recent[col + "_2020기준지수"] = (
        masan_recent[col] / base_value * 100
    )

display(masan_recent)

,연도,마산_수출실적(천달러),마산_수입실적(천달러),마산_무역수지(천달러),마산_고용인원,마산_업체수,마산_총교역액(천달러),마산_수출실적(천달러)_전년대비증가율,마산_수입실적(천달러)_전년대비증가율,마산_총교역액(천달러)_전년대비증가율,마산_고용인원_전년대비증가율,마산_업체수_전년대비증가율,마산_수출실적(천달러)_2020기준지수,마산_수입실적(천달러)_2020기준지수,마산_총교역액(천달러)_2020기준지수,마산_고용인원_2020기준지수,마산_업체수_2020기준지수
0,2020,768680.0,203814.0,564658.0,5328.0,120,972494.0,NaN,NaN,NaN,NaN,NaN,100.000000,100.000000,100.000000,100.000000,100.000000
1,2021,947255.0,238914.0,708341.0,5314.0,126,1186169.0,23.231384,17.221584,21.971858,-0.262763,5.000000,123.231384,117.221584,121.971858,99.737237,105.000000
2,2022,987303.0,228764.0,758539.0,5625.0,127,1216067.0,4.227795,-4.248391,2.520551,5.852465,0.793651,128.441354,112.241554,125.046221,105.574324,105.833333
3,2023,1056143.0,247410.0,808734.0,5662.0,131,1303553.0,6.972530,8.150758,7.194176,0.657778,3.149606,137.396966,121.390091,134.042267,106.268769,109.166667


핵심결과해석
1) 총 교역액 증가 (금액기준임, 양이 증가한 것은 아님), 23년에는 마산 자유무욕지역 총교역액은 20년 대비 약 34.0% 증가
2) 수출실적도 꾸준하게 증가, 23년 수출실적은 20년 대비 약 37.4% 증가 -> 마산 지유무역 지역에서 수출 기반 산업 활동이 확대되고 있단 근거로 사용 가능
3) 수입실적은 2022년에 잠깐 줄었지만 2023년에 다시 증가, 2022년에 전년 대비 감소하였지만 2023년에는 다시 8.15% 증가하여 회복 흐름보임
4) 업체 수 증가, 23년 업체수는 2020년 대비 약 9.2% 증가 -> 물류를 발생시키는 기업 수 늘은 의미로 볼 수 있음

=> 마산 자유무역지역은 20~23년 동안 수출실적, 총교역액, 업체 수 모두 증가, 이는 창원항만 배후 산업권역에서 수출입 기반 물류 수요가 지속적으로 발생하고 확대되고 있음을 보여주는 간접 지표로 해석 가능